### PR TITLE
Feature/api call

### LINKED ISSUE #4: [Feature]: Error handling
### Contact Details

_No response_

### New feature description?

When the server returns a 500 error, the user will see proper error handling

### How important is the requested feature?

Required

### LINKED ISSUE #5: [Feature]: PR Review
### Contact Details

_No response_

### New feature description?

As a web development student, when I submit a PR, I would like Kayla to review it, so I can get a good grade on this project.

### How important is the requested feature?

Required

### PR DESCRIPTION
**Pull request by nhodnett**
# Pull Request
It adds api call functionality and error handling functionality and styling.
## Review
@elleshadow Please check for any unused code, and any syntax errors we may have missed.
Kayla, please advise on sizing of poster in single movie view. Currently the image is warped in full screen view.
## Issues
1. close #5
2. close #4
## Type of change
- New feature (non-breaking change which adds functionality)
- Refactor feature (non-breaking change that modifies existing work)
- Cosmetic changes (changes to layout or visuals only)
## All Submission:
- [x] I completed all sections of this PR.
- [x] I have deleted unnecessary sections of this PR.
- [x] I have reviewed the rubric at the bottom of this PR (Even if deleting the section, it’s good to keep it all in mind)
- [x] I have added other team member as reviewers.
## New Feature Submissions:
- [x] New feature appears to be bug free.
- [x] When the project is run locally, the terminal shows no errors, warnings, or console.logs.
- [ ] Tests have been written for the change
- [ ] Change passes all tests

## Refactor Submissions:
- [] DRY: Code is as DRY as possible.
- [x] SRP: Each function/methods completes no more than 1 task.
- [x] Code follows the style guidelines of this project (semicolons, indentation, naming conventions, etc..).
- [x] All console logs have been removed from the code. (Seriously, users will never need them, get rid of them.)
- [x] All commented code has been removed, including commented console logs (commented explanations are acceptable)

### CODE REVIEW

**Review by elleshadow — APPROVED** [review]
Amazing work on this!

File: src/App.js [hunk 1]
```diff
@@ -1,53 +1,86 @@
...
 import './App.css';
 import movieDetails from './data/movieDetails';
-import moviesData from './data/movieData';
+import ErrorMessage from './components/ErrorMessage';
 
 class App extends Component{
   constructor() {
     super();
     this.state = {
       movieSelected: 0,
```
Comment (kaylagordon) [inline src/App.js:13]: You should be able to remove this piece of state after you've implemented Router

File: src/App.js [hunk 2]
```diff
@@ -1,53 +1,86 @@
...
       movieDetails: movieDetails,
-      movies: []
+      movies: [],
+      errorMessage: ''
     }
   }
 
+  componentDidMount = () => {
+    fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
```
Comment (kaylagordon) [inline src/App.js:21]: Consider moving your API calls to a separate file like you did in M2.

File: src/App.js [hunk 3]
```diff
@@ -1,53 +1,86 @@
...
+    .catch(error => {
+      this.setState({errorMessage: error.message})
+    })
+  }
+
   handleClick = (id) => {
-    // console.log(id)
-    this.setState({
-      movieSelected: id
+    {!!id ? fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id}`)
```
Comment (kaylagordon) [inline src/App.js:34]: Not sure the `!!` is needed here, but I might be missing something. Why couldn't `id ? fetch...` work?

File: src/App.js [hunk 4]
```diff
@@ -1,53 +1,86 @@
...
-        <Navbar 
+        <Navbar
             movieSelected={!!this.state.movieSelected}
             handleClick={this.handleClick}
         />
-
-        {this.state.movieSelected ? 
-          <MovieDetails 
+        {this.state.errorMessage ?
+        <ErrorMessage
```
Comment (kaylagordon) [inline src/App.js:74]: Nice use of an Error component

File: src/components/Navbar.js [hunk 5]
```diff
@@ -7,8 +7,10 @@ const Navbar = (props) => {
 
   return (
     <nav className='Navbar'>
-      {movieSelected && <button onClick={() => handleClick(0)}>back</button>}
-      <h1 className='siteHeader'>Rancid Tomatillos</h1>
+      <h1 className='siteHeader'onClick={() => handleClick(0)}>
+      <span className="rancid">Rancid</span>
+      <span className="tomatillos"> Tomatillos</span>
```
Comment (kaylagordon) [inline src/components/Navbar.js:12]: Fix indentation

File: src/styles/MovieCard.css [hunk 6]
```diff
@@ -1,13 +1,23 @@
+@import url('https://fonts.googleapis.com/css2?family=Roboto+Condensed:ital,wght@0,300;0,400;0,700;1,300;1,400;1,700&display=swap');
+
 .Card {
   width: 100%;
   background-color: #242322;
 }
 
+.Card:hover {
```
Comment (kaylagordon) [inline src/styles/MovieCard.css:8]: I love a good hover state! Makes apps feel more professional and polished.

File: src/App.js (hunk 1 above)
Comment (kaylagordon) [inline src/App.js:13]: It also looks like later in your code, this piece of state ends up being assigned `true` or `false`. I would recommend using a default value that is the same data type as what the assigned value would be later. So here, I would have the default value be `movieSelected: false`, not `0`. It won't change the functionality but it does make more sense for other devs coming into this code.

**Pull request merged**